src/protocols/XDGShell.cpp

```diff
@@ -534,30 +534,26 @@ CBox CXDGPositionerRules::getPosition(const CBox& constraint, const Vector2D& pa
     if (success)
         return predictedBox.translate(-parentCoord - constraint.pos());
 
+    CBox test = predictedBox;
+
     if (state.constraintAdjustment & (XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X | XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y)) {
         // attempt to flip
         const bool flipX = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X;
         const bool flipY = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y;
 
-        CBox       test = predictedBox;
-        success         = true;
-        if (flipX && test.copy().translate(Vector2D{-predictedBox.w - state.anchorRect.w, 0}).expand(-1).inside(constraint))
+        if (flipX && flipY)
```

Comment:
no, no, no. What if the flip gets us _away_ from the desired positioning?

Reply:
The only Idea I would have is to create a flipped copy and a regular copy, and then check if either of those provide a constraint that is ok.

But the translate check can't be within the if, otherwise the transform doesn't get applied and the slide won't do anything anymore. E.g. if a popup needs flip and transform it can't work with the translate within the if.

Reply:
how about test how many sides are out of bounds and flip if it decreases the count?

Reply:
Woops, sorry, you did the same with the other cases, so just check if horizontal out or vertical out.